subsys/net/lib/nrf_cloud/Kconfig.nrf_cloud_client_id

```diff
@@ -6,21 +6,15 @@
 menu "Client ID (nRF Cloud Device ID)"
 choice NRF_CLOUD_CLIENT_ID_SRC
 	prompt "Client ID Source"
-	default NRF_CLOUD_CLIENT_ID_SRC_IMEI
+	default NRF_CLOUD_CLIENT_ID_SRC_HW_ID
 
-config NRF_CLOUD_CLIENT_ID_SRC_IMEI
-	depends on NRF_MODEM_LIB
-	bool "IMEI based value"
+config NRF_CLOUD_CLIENT_ID_SRC_HW_ID
```

Comment:
The point in making this library is to remove the modem dependency from libraries like `nrf_cloud`. If the library just explicitly selects its ID itself, there is little benefit of using the `hw_id` library at all.

> in general, i don't understand why the HW_ID_LIBRARY_SOURCE choice must impact the choice of a device's identity with a cloud.

My premise is that this is the main purpose of the option. An IoT application doesn't have much use for multiple different unique IDs on one particular device, right?

Reply:
After having another look, I think I'll reduce the changes to the `nrf_cloud` library and just provide hw_id as another input option.

Reply:
```
An IoT application doesn't have much use for multiple different unique IDs on one particular device, right?
```
I would not make that assumption.  Why restrict the application to have access to only one id?  Maybe one service requires an IMEI and another service requires a something else. 

```
The point in making this library is to remove the modem dependency from libraries like nrf_cloud. If the library just explicitly selects its ID itself, there is little benefit of using the hw_id library at all.
```

The modem dependency is still there if the IMEI is used.
Having a common library to obtain the various hardware id values is beneficial imo.

Reply:
> I would not make that assumption. Why restrict the application to have access to only one id? Maybe one service requires an IMEI and another service requires a something else.

I think that the application can go the direct way of accessing those IDs if it really needs multiple different ones. This library is only a thin wrapper around the specific access functions anyway. It's nice to be able to quickly configure the cloud libraries' client ID with this, but in other use cases the difference is mostly in function name.

> The modem dependency is still there if the IMEI is used.

That's true. The more pressing issue of compiling nrf cloud without a modem is probably already solved.

Reply:
Isn't the choice of ID more cloud-solution-specific, and therefore specific to both the combination of cloud solution and application, rather than specific to the app alone?

Like, what if the application needs to use two different libraries that both want to rely on `hw_id`, and in this case, the application needs Library A to use source A, and needs Library B to use source B? If Library A and B are both aware of this possibility, they will (wisely) ignore the internal configuration of `hw_id` and provide their own configuration, as is essentially the case with `nrf_cloud` in your latest revision.

That means that any library that cares about being fully modular is not going to be able to use `hw_id`'s internal configuration. And taking your above recommendation as fact, that means those libraries should not use `hw_id` at all, but should access the IDs themselves directly.

Personally, I agree with Justin that `hw_id` is at least a convenient way to get access to a wide variety of IDs.

In fact (and I'm getting waaayyy outside the scope of your PR, here, at this point, so feel free to ignore:) It might be even more useful if it provided:

- A struct representing the available ID types, including an extra type, "custom"
- Some macros to make selecting struct values by name in KConfig easier
- A function to get the ID corresponding to a value of the struct, with a default-value-pointer argument that is returned if "custom" is selected

Then library designers could rely on `hw_id` to easily implement the hardware ID selection **_pattern_** without fear of interference.

Reply:
> Isn't the choice of ID more cloud-solution-specific, and therefore specific to both the combination of cloud solution and application, rather than specific to the app alone?

I would say this is a decision you make when you implement cloud provisioning. Still, there might be conventions forcing different IDs to be used in a single application. I'll create a future-enhancement task to expand the API of `hw_id` later.